Update GitHub action updating used WordPress and plugins [MAILPOET-6097]

lysyjan opened the pull request:
## Description

This PR fixes and improves GH action for updating used WordPress images and downloaded plugins:

- fixed using env variable for the step updating Automate Woo
- added saving versions to temporary files
- added step opening pull-request
- added comments
- used function for updating version based on tags from GH private repositories to avoid code repetition
- used a specific version for WooCommerce instead of the latest for unification with the rest of the plugins

## Code review notes

This is almost the final step. I would like to execute this action on the trunk, and if everything is fine, we can add a cron definition.

## QA notes

We don't need QA here.

## Linked PRs

https://github.com/mailpoet/mailpoet/pull/5659
https://github.com/mailpoet/mailpoet/pull/5660
https://github.com/mailpoet/mailpoet/pull/5661

## Linked tickets

[MAILPOET-6097]

## After-merge notes

_N/A_

## Tasks

- [x] 🚫 I followed [best practices](https://codex.wordpress.org/I18n_for_WordPress_Developers) for translations
- [x] 🚫 I added sufficient test coverage
- [x] 🚫 I embraced TypeScript by either creating new files in TypeScript or converting existing JavaScript files when making changes


[MAILPOET-6097]: https://mailpoet.atlassian.net/browse/MAILPOET-6097?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

lysyjan committed:
Move env variable definition to the correct step

[MAILPOET-6097]

lysyjan committed:
Change branch name for the pushed changes

[MAILPOET-6097]

lysyjan committed:
Add step creating pull-request

[MAILPOET-6097]

lysyjan committed:
Rename file containing update workflow

[MAILPOET-6097]

lysyjan committed:
Add used WordPress versions to the commit message

[MAILPOET-6097]

lysyjan committed:
Use exact version instead of latest

To unify with other plugins I decided to use the exact version number.
[MAILPOET-6097]

lysyjan committed:
Update script getting WooCommerce versions and store them to file

[MAILPOET-6097]

lysyjan committed:
Extract replacing versions for a private plugin to a function

[MAILPOET-6097]

lysyjan committed:
Add used plugin versions to the commit messages

[MAILPOET-6097]

costasovo commented on .github/workflows/check-wordpress-versions.yml:
```diff
@@ -127,3 +127,13 @@ jobs:
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+          branch: update-plugins-and-wordpress
+          title: 'Update WordPress Docker and WooCommerce Versions'
```
Afaik, it is not only WooCommerce, but it could also be other plugins, right?

lysyjan replied:
This title is a bit tricky because it can be all plugins without WP or only WC, etc.
Maybe Porter can update it. We could generate it, but I am not sure if it is worth it.

costasovo replied:
It is probly not worth it. I would keep it as is. In case we find it confusing we can use something more generic like "update versions in Circle CI configuration"

costasovo commented on .github/workflows/scripts/check_wordpress_versions.php:
```diff
@@ -46,15 +61,21 @@ function getLatestAndPreviousVersions($sortedVersions) {
+/**
+ * We specify the latest WordPress version only in the docker-compose file for the tests.
+ */
+function replaceLatestVersion($latestVersion): void {
```
I see that you don't use typehints for method parameters. Here but also for other functions. Is that on purpose? I think they can be helpful to catch error when something (e.g. API response) changes.

costasovo reviewed (commented):
@lysyjan Thanks for adding the comments to the functions! It helps to figure out what a function does. Overall looks good! Great job!

I added just a couple of comments. Small details.

lysyjan committed:
Add type declarations to function in GH Action update scripts

This commit also fixes some bugs with function redeclaration and missing parameters.
[MAILPOET-6097]

lysyjan committed:
Update title for created pull request via GH action

[MAILPOET-6097]

lysyjan commented:
@costasovo Thank you for your feedback. I added two new commits addressing your suggestions.
